marker/src/main/java/App.java

```diff
@@ -46,22 +43,10 @@
    * @param args command line args
    */
   public static void main(String[] args) {
-
-    final Logger logger = LoggerFactory.getLogger(App.class);
-    Guard guard = new Guard();
-    Thief thief = new Thief();
-
-    if (guard instanceof Permission) {
-      guard.enter();
-    } else {
-      logger.info("You have no permission to enter, please leave this area");
-    }
-
-    if (thief instanceof Permission) {
-      thief.steal();
-    } else {
-      thief.doNothing();
-    }
```

Comment:
The point of the design pattern is to demonstrate marker interfaces, so I think you cannot delete the `instanceof` checks here

Reply:
Pushed the changes